include/solvers/heat_transfer.h

```diff
@@ -56,12 +60,36 @@ class HeatTransfer : public AuxiliaryPhysics<dim, TrilinosWrappers::MPI::Vector>
     , triangulation(p_triangulation)
     , simulation_control(p_simulation_control)
     , dof_handler(*triangulation)
-    , fe(simulation_parameters.fem_parameters.temperature_order)
     , solution_transfer(dof_handler)
     , solution_transfer_m1(dof_handler)
     , solution_transfer_m2(dof_handler)
     , solution_transfer_m3(dof_handler)
-  {}
+  {
+    if (simulation_parameters.mesh.simplex)
+      {
+        // for simplex meshes
+        fe = std::make_shared<Simplex::FE_P<dim>>(
+          simulation_parameters.fem_parameters.temperature_order);
+        temperature_mapping = std::make_shared<MappingFE<dim>>(*fe);
+        cell_quadrature =
+          std::make_shared<Simplex::QGauss<dim>>(fe->degree + 1);
+        face_quadrature =
+          std::make_shared<Simplex::QGauss<dim - 1>>(fe->degree + 1);
+        error_quadrature =
+          std::make_shared<Simplex::QGauss<dim>>(fe->degree + 2);
```

Comment:
With the new changes this would be something like: 

```suggestion
        fe = std::make_shared<FE_SimplexP<dim>>(
          simulation_parameters.fem_parameters.temperature_order);
        temperature_mapping = std::make_shared<MappingFE<dim>>(*fe);
        cell_quadrature =
          std::make_shared<QGaussSimplex<dim>>(fe->degree + 1);
        face_quadrature =
          std::make_shared<QGaussSimplex<dim - 1>>(fe->degree + 1);
        error_quadrature =
          std::make_shared<QGaussSimplex<dim>>(fe->degree + 2);
```